### PR TITLE
Sdl rendering

### PR DESCRIPTION
**Pull request by Emiluren**

### CODE REVIEW

**Review by HannesTuhkala — APPROVED** [review]

**Review by HannesTuhkala — CHANGES_REQUESTED** [review]
Not tested on Windows yet

File: Cargo.toml [hunk 1]
```diff
@@ -8,13 +8,16 @@ build = "build.rs"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-ggez = "0.5"
+serde = "1.0.102"
+serde_derive = "1.0.102"
```
Comment (TheZoq2) [inline Cargo.toml]: Are these used in the client/server again? Previously the were only used in libplen

File: Cargo.toml [hunk 2]
```diff
@@ -8,13 +8,16 @@ build = "build.rs"
...
 bincode = "1.2.0"
 nalgebra = {version = "0.18.1", features = ["serde-serialize"]}
 rand = "0.7.2"
 rand_derive = "0.5.0"
 whoami = "0.6.0"
-sdl2 = { version = "0.32.2", features = ["mixer"] }
+strum = "0.16.0"
+strum_macros = "0.16.0"
```
Comment (TheZoq2) [inline Cargo.toml]: Same thing with strum

File: libplen/src/constants.rs [hunk 3]
```diff
@@ -3,7 +3,7 @@ use std::f32::consts::PI;
 pub const PLANE_SIZE: u32 = 20;
 pub const BULLET_RADIUS: u32 = 1;
 pub const POWERUP_RADIUS: u32 = 20;
-pub const POWERUP_SPEED: f32 = 0.03;
+pub const POWERUP_SPEED: f32 = 1.8;
```
Comment (TheZoq2) [inline libplen/src/constants.rs:6]: #83 changed this to 1.5

File: src/map.rs [hunk 4]
```diff
@@ -263,283 +214,271 @@ impl Map {
...
-    fn draw_killfeed(ctx: &mut ggez::Context, assets: &Assets, game_state: &GameState) {
-        let mut i: f32 = 0.;
-        let mut kf = game_state.killfeed.clone();
-        let messages = kf.get_messages().clone();
+    fn draw_killfeed(canvas: &mut Canvas<Window>, assets: &Assets, game_state: &GameState) -> Result<(), String> {
+        let mut i = 0;
+        let mut kill_feed = game_state.killfeed.clone();
+        let messages = kill_feed.get_messages().clone();
+
         for message in messages.iter() {
```
Comment (TheZoq2) [inline src/map.rs]: use `for (i, message) = messages.iter().enumerate()` instead of `mut i = 0`

File: src/menu.rs [hunk 5]
```diff
@@ -27,148 +24,137 @@ const COLORS: [player::Color; 5] = [
...
+        );
+
+        canvas.set_draw_color((0, 0, 0, 128));
+        canvas.fill_rect(background_rect)?;
+
+        let texture_creator = canvas.texture_creator();
+
+        let text = assets.font.render(
+            self.plane.name()
+        ).blended((255, 255, 255)).expect("Could not render text");
```
Comment (TheZoq2) [inline src/menu.rs]: Weird indentation, I would prefer

```rust
assets.font.render(self.plane_name())
    .blended((255, 255, 255))
    .expect("Could not render text")
```

File: src/menu.rs [hunk 6]
```diff
@@ -27,148 +24,137 @@ const COLORS: [player::Color; 5] = [
...
+        let text_texture = texture_creator.create_texture_from_surface(text).unwrap();
+        draw_texture(canvas, &text_texture, na::Point2::new(px + 10., py + 10.))?;
+
+        let instruction = assets.font.render(
+            "click to change plane blyat:"
+        ).blended((255, 255, 255)).expect("Could not render text");
```
Comment (TheZoq2) [inline src/menu.rs]: Same thing here

File: src/menu.rs [hunk 7]
```diff
@@ -27,148 +24,137 @@ const COLORS: [player::Color; 5] = [
...
+            cy as i32,
+            constants::COLOR_SELECTION_SIZE as u32,
+            constants::COLOR_SELECTION_SIZE as u32
+        );
+        canvas.set_draw_color(self.color.rgba());
+        canvas.fill_rect(background_rect)?;
+
+        let instruction = assets.font.render(
+            "click to change color:"
+        ).blended((255, 255, 255)).expect("Could not render text");
```
Comment (TheZoq2) [inline src/menu.rs]: And here

**Review by TheZoq2 — CHANGES_REQUESTED** [review]
Looks good apart from a few style issues

File: Cargo.toml (hunk 1 above)
Comment (Emiluren) [reply Cargo.toml]: Oops, must have slipped in with the merge conflict

File: libplen/src/constants.rs (hunk 3 above)
Comment (Emiluren) [reply libplen/src/constants.rs:6]: You're thinking of the wrong value POWERUP_SPEED_BOOST, not POWERUP_SPEED

File: src/menu.rs (hunk 5 above)
Comment (Emiluren) [reply src/menu.rs]: Yes, I was thinking that we would need to cache the rendered strings for performance reasons so I just made some throwaway code but then it seemed to be fine performance wise. Will fix

File: src/map.rs (hunk 4 above)
Comment (Emiluren) [reply src/map.rs]: I wasn't the one who wrote this code originally but sure, I can fix it :P

**Review by TheZoq2 — APPROVED** [review]

**Review by TheZoq2 — APPROVED** [review]

**Review by malcx95 — APPROVED** [review]

**Review by malcx95 — CHANGES_REQUESTED** [review]
Name tags are no longer centered. It would be nice if you fixed this.

**Review by malcx95 — APPROVED** [review]

**Review by HannesTuhkala — APPROVED** [review]
Works on Windows

**Pull request merged**